lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -268,10 +236,10 @@ sub _show {
     my $rd = $job->result_dir();
     if ($rd) {    # saved anything
                   # result files box
-        my @resultfiles = test_resultfile_list($job->result_dir());
+        my @resultfiles = $job->test_resultfile_list;
 
         # uploaded logs box
-        my @ulogs = test_uploadlog_list($job->result_dir());
+        my @ulogs = $job->test_uploadlog_list;
```

Comment:
I guess if everyone wants to pass the result as reference, then possibly the functions should return exactly that. Now we cast it in two places

Reply:
Done :)